src/controllers/userController.js

```diff
@@ -1,59 +1,36 @@
-    expiresIn: process.env.JWT_LIFETIME,
-  });
-  return token;
-};
-
-//returning a regular javascript object that is a subset of the fields of the User model.
-const formatUserForFrontend = (user) => {
-  return { name: user.name, userId: user._id, role: user.role };
-};
-
-
-const getSingleUser = async (req, res) => {
-  const user = await User.findOne({ _id: req.params.id });
-  if (!user) {
-    res.status(400).json({ msg: `No user with id : ${req.params.id}` });
-  }
-  // If the user is found, it calls the formatUserForFrontend function to format the user data before sending it in the response
-  const tokenUser = formatUserForFrontend(user);
-
-  res.status(200).json({ user : tokenUser });
+const { StatusCodes } = require('http-status-codes');
+const CustomError = require('../errors');
+const { createTokenUser, attachCookiesToResponse, checkPermissions } = require('../utils');
+
+const getSingleUser = async(req,res) => { // Function to get a single user by ID
+    // Find the user in the database based on the provided user ID and exclude the 'password' field
+    const user = await User.findOne({ _id: req.params.id }).select('-password');
+    if(!user) { // Throw a NotFoundError if the user is not found
+        throw new CustomError.NotFoundError(`No user with id: ${req.params.id}`);
+    }
+    checkPermissions(req.user, user._id); // Check if the user has permission to access the user's information
+    res.status(StatusCodes.OK).json({ user }); // Send a JSON response with the status code 200 OK and the user
 };
 
-const updateUser = async (req, res) => {
-  const { email, name } = req.body;
-  if (!email && !name) {
-    res.status(400).json({ msg: 'Please provide all values' });
-  }
-
-
-  const user = await User.findOne({ _id: req.params.id }); //get user
-  // Update email if request included a non-null value, otherwise keep the email as-is
-  user.email = email || user.email;
-
-  // Update name if request included a non-null value, otherwise keep the name as-is
-  user.name = name || user.name;
-
-  await user.save(); // using pre save hook (see user model) to save updated user
-  //create tokenUser with data updated
-  const createJWT = (user) => {
-    return { name: user.name, userId: user._id, role: user.role };
-  };
-
-
-  user = await user.save();
-  //After updating the user, it calls the formatUserForFrontend function to format the updated user data before sending it in the response
-  const tokenUser = formatUserForFrontend(user);
-  res.status(200).json({ user: tokenUser });
+// update user with user.save()
+const updateUser = async(req,res) => {
+    const { email, name } = req.body;
+    if (!email || !name) { // Check if email and name are provided
+        throw new CustomError.BadRequestError('Please provide all values');
+    }
+    const user = await User.findOne({ _id: req.user.userId }); // Find the user in the database based on the current user's ID
```

Comment:
This means this endpoint is only ever for updating the _requesting user_ info. Is that the intention? Will we ever want for example the ability for an admin to edit user information? If this is the intention, it may be more clear to add a comment up top or name the function something like `updateCurrentUser`